core/templates/dev/head/pages/topic_editor/questions/QuestionsTabDirective.js

```diff
@@ -231,8 +232,11 @@ oppia.directive('questionsTab', [
               backdrop: 'static',
               keyboard: false,
               controller: [
```

Comment:
There seems to be a good deal of duplication between the question editor in the skill/topic editor. Would it be better to consolidate these? For time's sake, I'm also okay with just filing an issue and deferring it.

Reply:
Just a note to please try and consolidate if possible, since duplication in the codebase makes maintenance much harder later on (you always have to do new changes twice, and in fact it introduces the risk of skew between the two versions). It gets even worse if folks then take the existing duplication to indicate that duplication is OK and it's fine to copy/paste to yet another version; this is something I've seen happen recently. So best to eliminate duplication at the source, although that's usually harder than copy/pasting.

Reply:
Ok, will try to deduplicate this.
Shall I try creating a questions tab directive which both these editors can access?

Reply:
That sounds reasonable to me if there's pretty much no "branching" (i.e. the directive operates similarly in both cases). Otherwise you may need some light configuration via attrs and that's fine (as long as it's not a full "if in mode X, do this, else do something completely different"). If heavy configuration is needed then -- we might need to discuss.

Reply:
Done, created a new questions-list directive, that shares all overlapping functionalities between topic and skill editor.